DomainContent/FiresideChat/buttons/todayButton.js

```diff
@@ -0,0 +1,155 @@
+        },
+
+        getButtonTarget: function() {
+            if (properties.name.indexOf("Web Entities") !== NEGATIVE) {
+                return "Web";
+            } else {
+                return "Backdrop";
+            }
+        },
+
+        pressButton: function() {
+            if (_this.type === "Raise") {
+                _this.lowerButton();
+                if (_this.target === "Web") {
+                    windows.forEach(function(element) {
+                        Entities.callEntityServerMethod(element, 'raise');
+                    });
+                } else {
+                    Entities.callEntityServerMethod(backdrop, 'raise');
+                }
+                Script.setTimeout(function() {
+                    _this.raiseButton();
+                }, DOWN_TIME_MS);
+            } else {
+                _this.lowerButton();
+                if (_this.target === "Web") {
+                    windows.forEach(function(element) {
+                        Entities.callEntityServerMethod(element, 'lower');
+                    });
+                } else {
+                    Entities.callEntityServerMethod(backdrop, 'lower');
+                }
+                Script.setTimeout(function() {
+                    _this.raiseButton();
+                }, DOWN_TIME_MS);
+                return;
+            }
+        },
+
+        raiseButton: function() {
+            Entities.editEntity(_this.entityID, {
+                position: properties.position
+            });
+        },
+
+        lowerButton: function() {
+            var HAPTIC_STRENGTH = 1;
+            var HAPTIC_DURATION = 20;
+            Controller.triggerHapticPulse(HAPTIC_STRENGTH, HAPTIC_DURATION, currentHand);
+            properties.position.y -= BUTTON_PRESS_OFFSET;
+            Entities.editEntity(_this.entityID, {
+                position: properties.position
+            });
+            properties.position.y += BUTTON_PRESS_OFFSET;
+        },
+        
+        mousePressOnEntity: function(entityID, mouseEvent) {
+            if (!mouseEvent.button === "Primary") {
+                return;
+            }
+            if (!Pointers.isMouse(mouseEvent.id)) {
+                if (Pointers.isLeftHand(mouseEvent.id)) {
+                    currentHand = 0;
+                } else if (Pointers.isRightHand(mouseEvent.id)) {
+                    currentHand = 1;
+                }
+            }
+            _this.pressButton();
+        },
+
+        startNearTrigger: function(entityID, mouseEvent) {
+            if (Pointers.isLeftHand(mouseEvent.id)) {
+                currentHand = 0;
+            } else if (Pointers.isRightHand(mouseEvent.id)) {
+                currentHand = 1;
+            }
+            _this.pressButton();
+        },
+
+        stopNearTrigger: function(entityID, mouseEvent) {
+            if (!mouseEvent.button === "Primary") {
+                return;
+            }
+            if (!Pointers.isMouse(mouseEvent.id)) {
```

Comment:
What is the logic behind tracking the hand? To avoid duplicate events?

Reply:
That is for the haptic feedback